### PR TITLE
me.timschneeberger.jdsp4linux

### PR DESCRIPTION
**Pull request by ThePBone**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*

~This application has two flavors: one compiled for PulseAudio (`me.timschneeberger.jdsp4linux.pulse`) and one for PipeWire (`me.timschneeberger.jdsp4linux.pipewire`). Because of that, they need to have ownership of the shared D-Bus service name `--own-name me.timschneeberger.jdsp4linux`.
It would be impractical for 3rd-party developers having to deal with two different service names depending on which flavor is installed.~

~Both of the flavors clarify this in their metainfo XMLs like this:~
```xml
<provides>
    <binary>jamesdsp</binary>
    <dbus type="user">me.timschneeberger.jdsp4linux</dbus>
</provides>
```
~This submission contains the PipeWire flavor. The submission of the PulseAudio flavor can be found in PR #4251.~

- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/Audio4Linux/JDSP4Linux
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by ThePBone** [comment]
bot build, me.timschneeberger.jdsp4linux.pipewire

**Comment by ThePBone** [comment]
bot, build me.timschneeberger.jdsp4linux.pipewire

**Comment by ThePBone** [comment]
bot, build me.timschneeberger.jdsp4linux.pulse

**Comment by hfiguiere** [comment]
but do you really need two apps? How is one chosen over the over. Also pipewire somewhat unifies everything so is a pulse variant necessary?

**Comment by ThePBone** [comment]
> but do you really need two apps? How is one chosen over the over. Also pipewire somewhat unifies everything so is a pulse variant necessary?

Yes. This is a desktop audio effects processor (like EasyEffects) that needs to target advanced Pulse/PipeWire APIs directly. 
There are no unified APIs for this particular use case. 
The user chooses a flavor based on whether they installed PipeWire or PulseAudio on their system.

Pipewire does have a Pulseaudio compatibility mode, however, it doesn't re-implement the particular APIs that I need.

**Comment by ThePBone** [comment]
bot, build me.timschneeberger.jdsp4linux.pipewire

**Comment by hfiguiere** [comment]
> The user chooses a flavor based on whether they installed PipeWire or PulseAudio on their system.

If you run this as a flatpak, there is pipewire, and I don't think anything else matter. Otherwise it's pretty poor UX to have to select a different app due to an implementation detail.

**Comment by ThePBone** [comment]
>If you run this as a flatpak, there is pipewire, and I don't think anything else matter. Otherwise it's pretty poor UX to have to select a different app due to an implementation detail.

No, it needs to access the host installation of Pulse or Pipewire. It can't do that part in the sandbox, because it is supposed to intercept and manipulate the audio stream of other native applications on the host system.

I could drop PulseAudio support altogether as EasyEffects did for their app; however, I want to avoid that because some users depend on Pulse support.

**Comment by ThePBone** [comment]
Change of plans: Since the PulseAudio version is legacy stuff, I'll only submit the new PipeWire build (under a new app id), which is the recommended version anyway.

This is a good opportunity to push users that are still sticking to the legacy pulseaudio implementation to the new pipewire-focused implementation that has much better performance.

**Comment by hfiguiere** [comment]
you didn't need to close this.

**Comment by ThePBone** [comment]
bot, build me.timschneeberger.jdsp4linux

File: me.timschneeberger.jdsp4linux.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - "*.m4"
+  - "*.pm"
+  - /include
+  - /lib/pkgconfig
+  - /share/pkgconfig
+modules:
+  - name: jamesdsp
+    buildsystem: qmake
+    modules:
+      - jamesdsp-modules.yaml
```
Comment (hfiguiere) [inline me.timschneeberger.jdsp4linux.yaml:28]: the *content* of this should be inserted above line 25.

It's more maintainable.

**Comment by ThePBone** [comment]
bot, build me.timschneeberger.jdsp4linux

**Pull request closed without merging**